signer/src/bitcoin/validation.rs

```diff
@@ -91,10 +100,117 @@ pub fn is_unique(package: &[TxRequestIds]) -> bool {
 
 impl BitcoinPreSignRequest {
     /// Validate the current bitcoin transaction.
-    pub fn pre_validation(&self) -> Result<(), Error> {
+    async fn pre_validation<C>(
+        &self,
+        ctx: &C,
+        btc_ctx: &BitcoinTxContext,
+    ) -> Result<ValidationCache, Error>
+    where
+        C: Context + Send + Sync,
+    {
         if !is_unique(&self.request_package) {
             return Err(Error::DuplicateRequests);
         }
+        let cache = self.fetch_all_reports(ctx, btc_ctx).await?;
+
+        self.validate_max_mintable(ctx, &cache).await?;
+
+        Ok(cache)
+    }
+
+    async fn fetch_all_reports<C>(
+        &self,
+        ctx: &C,
```

Comment:
suggestion: looks like you could just take a `DbRead` here instead of a full `Context`

Reply:
done in https://github.com/stacks-network/sbtc/pull/811/commits/086c1a9a380d8416c63ea1edca2d3307453a0bf1